gateway/ln-gateway/src/lightning/lnd.rs

```diff
@@ -1459,11 +1459,11 @@ mod tests {
         let features_payment_secret = {
             let mut f = Bolt11InvoiceFeatures::empty();
             f.set_payment_secret_optional();
-            WithoutLength(&f).encode()
+            f.encode()
         };
         assert_eq!(
             wire_features_to_lnd_feature_vec(&features_payment_secret).unwrap(),
-            vec![15]
+            vec![15, 17]
```

Comment:
Looks sus, why did the test change?

Reply:
Fixed! This was needed previously because we were also encoding the length.